frontend/src/concepts/pipelines/topology/PipelineVisualizationSurface.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import {
+  action,
+  createTopologyControlButtons,
+  DEFAULT_SPACER_NODE_TYPE,
+  defaultControlButtonsOptions,
+  getEdgesFromNodes,
+  PipelineNodeModel,
+  TopologyControlBar,
+  TopologyView,
+  useVisualizationController,
+  VisualizationSurface,
+} from '@patternfly/react-topology';
+import { useDeepCompareMemoize } from '~/utilities/useDeepCompareMemoize';
+
+type PipelineVisualizationSurfaceProps = {
+  nodes: PipelineNodeModel[];
+};
+
+const PipelineVisualizationSurface: React.FC<PipelineVisualizationSurfaceProps> = ({
+  nodes: unsafeNodes,
+}) => {
+  const nodes = useDeepCompareMemoize(unsafeNodes);
+  const controller = useVisualizationController();
+
+  React.useEffect(() => {
+    const edges = getEdgesFromNodes(
+      nodes.filter((n) => !n.group),
+      DEFAULT_SPACER_NODE_TYPE,
+    );
+
+    controller.fromModel(
+      {
+        nodes,
+        edges,
+      },
+      true,
+    );
+    controller.getGraph().layout();
+  }, [controller, nodes]);
+
+  return (
+    <TopologyView
+      controlBar={
+        <TopologyControlBar
```

Comment:
Tooltip in here needs `removeFindDomNode`, but we cant control thay

Reply:
Going away in PF v5. [Slack thread](https://patternfly.slack.com/archives/C4FM977N0/p1679944054180079?thread_ts=1679943976.195919&cid=C4FM977N0). No-Op.